plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -0,0 +1,191 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { TableRow } from '@woocommerce/components/build-types/table/types';
+import { Icon, plugins } from '@wordpress/icons';
+import { gmdateI18n } from '@wordpress/date';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Subscription } from '../../types';
+import StatusPopover from './status-popover';
+import ActivationToggle from './activation-toggle';
+import ActionsDropdownMenu from './actions-dropdown-menu';
+
+function getStatus( subscription: Subscription ): {
+	text: string;
+	warning: boolean;
+	explanation?: string;
+} {
+	// TODO add statuses for subscriptions
+	if ( subscription.active ) {
+		return {
+			text: __( 'Active', 'woocommerce' ),
+			warning: false,
+		};
+	}
+
+	if ( subscription.product_key === '' ) {
+		return {
+			text: __( 'Not found', 'woocommerce' ),
+			warning: true,
+			explanation: __( 'This subscription is not found.', 'woocommerce' ),
+		};
+	}
+
+	if ( subscription.expired ) {
+		return {
+			text: __( 'Expired', 'woocommerce' ),
+			warning: true,
+			explanation: __(
+				'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+				'woocommerce'
+			),
+		};
+	}
+
+	return {
+		text: __( 'Inactive', 'woocommerce' ),
+		warning: true,
+		explanation: __(
+			'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+			'woocommerce'
+		),
+	};
+}
+
+function getVersion( subscription: Subscription ): string {
+	if ( subscription.local.version === subscription.version ) {
+		return subscription.local.version;
+	}
+
+	if ( subscription.local.version && subscription.version ) {
+		return subscription.local.version + ' > ' + subscription.version;
+	}
+
+	if ( subscription.version ) {
+		return subscription.version;
+	}
+
+	if ( subscription.local.version ) {
+		return subscription.local.version;
+	}
+
+	return '';
+}
+
+export function productName( subscription: Subscription ): TableRow {
+	// This is the fallback icon element with products without an icon.
+	let iconElement = <Icon icon={ plugins } size={ 40 } />;
+
+	// If the product has an icon, use that instead.
+	if ( subscription.product_icon ) {
+		iconElement = (
+			<img
+				src={ subscription.product_icon }
+				alt={ sprintf(
+					/* translators: %s is the product name. */
+					__( '%s icon', 'woocommerce' ),
+					subscription.product_name
+				) }
+				className="woocommerce-marketplace__my-subscriptions__product-icon"
+			/>
+		);
+	}
+
+	const displayElement = (
+		<div className="woocommerce-marketplace__my-subscriptions__product">
+			{ iconElement }
+			<span className="woocommerce-marketplace__my-subscriptions__product-name">
```

Comment:
These CSS classnames are really getting out of control! I don't think we're using BEM quite as it's designed, actually. But we can improve this another time.

Reply:
Hmmm, maybe it should be just `woocommerce-marketplace__product-name`. Do you mean something like that?

Reply:
I believe the original intention of `block__element--modifier` is that `block` is a fairly small object. Whereas we've been using `woocommerce-marketplace`, which represents the whole feature. So we end up with long chains like this. To address this properly needs a more careful rethink later!